components/automate-ui/src/app/pages/project/rules/project-rules.component.spec.ts

```diff
@@ -0,0 +1,219 @@
+    });
+
+    it('should have a backroute to project page', () => {
+      const backRoute = ['/settings', 'projects', component.project.id];
+      const compBackRoute = component.backRoute();
+      expect(backRoute).toEqual(compBackRoute);
+    });
+
+    it('the formgroup should have one condition', () => {
+      const conditions = component.ruleForm.get('conditions').value.length;
+      expect(conditions).toBe(1);
+    });
+
+    it('should create a condition', () => {
+      component.addCondition();
+      const conditions = component.ruleForm.get('conditions').value.length;
+      expect(conditions).toBe(2);
+    });
+
+    it('should delete a condition', () => {
+      let conditions = component.ruleForm.get('conditions').value.length;
+      expect(conditions).toBe(1);
+      component.deleteCondition(0);
+      conditions = component.ruleForm.get('conditions').value.length;
+      expect(conditions).toBe(0);
+    });
+
+    it('should return condition value of string', () => {
+      const conditionValue = component.getConditionValue('adasd');
+      expect(typeof conditionValue === 'string').toBeTruthy();
+    });
+
+    it('should return condition value of array', () => {
+      const conditionValue = component.getConditionValue(['adasd']);
+      expect(typeof conditionValue === 'string').toBeTruthy();
+    });
+
+    it('form should be invalid', () => {
+      expect(component.ruleForm.valid).toBeFalsy();
+    });
+
+    it('should enable submit when valid', () => {
+      component.ruleForm.get('name').setValue('My Rule');
+      component.ruleForm.get('type').setValue('NODE');
+      component.ruleForm.get('conditions').setValue([
+        {
+          attribute: 'CHEF_ORGS',
+          operator: 'EQUAL',
+          values: 'my value'
+        }
+      ]);
+      expect(component.ruleForm.valid).toBeTruthy();
+    });
+
+    it('should have attribute label with NODE type', () => {
+      component.ruleForm.get('type').setValue('NODE');
+      const attributeLabel = component.getAttributeLabel();
+      expect(attributeLabel).toBe('node attribute');
+    });
+
+    it('should have attribute label with EVENT type', () => {
+      component.ruleForm.get('type').setValue('EVENT');
+      const attributeLabel = component.getAttributeLabel();
+      expect(attributeLabel).toBe('event attribute');
+    });
+
+    it('should not show add label with one condition', () => {
+      const showAndLabel = component.showAndLabel(0);
+      expect(showAndLabel).toBeFalsy();
+    });
+
+    it('should not show delete button with one condition', () => {
+      const showDelete = component.showDelete(0);
+      expect(showDelete).toBeFalsy();
+    });
+
+    it('should show add label with two conditions', () => {
+      component.addCondition();
+      const showAndLabel = component.showAndLabel(0);
+      expect(showAndLabel).toBeTruthy();
+    });
+
+    it('should show delete button with two conditions', () => {
```

Comment:
```suggestion
    it('should show delete button on first condition with two conditions', () => {
```

Then should add one more test:
```
should show delete button on second condition with two conditions
```
(This correlates to @susanev 's comment about the delete on the last row needs to appear.)